File: ShowCase/src/app/Items/Spell.ts
import {Base, Price} from "./Base";
import {SpellName} from "./Potion";


export class CasterPossibility {
  constructor(name: string, level:number) {
  }
}

export class Spell extends Base {


  constructor(public id: number = 0,
              public name: string = "",
              public description: string = "",
              public source: string = "",
              public price: Price = new Price(),
              public casterPossibility: Array<CasterPossibility> = new Array<CasterPossibility>(),
              public levell: number = 0
  ) {
    super(id, name, description, source, price);
  }
}

File: OccultMerchant/SecondaryLocation/Items/Spell.cs
using System;

namespace SecondaryLocation.Items
{
    public class Spell : Item,ISpell
    {
        public Guid id { get; set; }
        public string name { get; set; }
        public string description { get; set; }
        public string source { get; set; }
        public int price { get; set; }
        public int ItemType { get; set; }
        public int range { get; set; }
        public string target { get; set; }
        public string duration { get; set; }
        public string savingThrow { get; set; }
        public bool spellResistence { get; set; }
        public string casting { get; set; }
        public string component { get; set; }
        public string school { get; set; }
        public string level { get; set; }
    }
}